File: fixtures/adversarial/T6/go/silent/after/format_test.go
package demo

import (
	"errors"
	"testing"
)

func TestRefusesAnEmptyInput(t *testing.T) {
	if _, err := Parse(""); !errors.Is(err, ErrEmpty) {
		t.Errorf("parsing an empty input let it through")
	}
}

func TestRefusesAStraySeparator(t *testing.T) {
	if _, err := Parse(";;"); err.Error() != "stray separator" {
		t.Errorf("parsing a stray separator let it through")
	}
}

func TestRefusesATrailingSeparator(t *testing.T) {
	if _, err := Parse("a;"); !errors.Is(err, ErrTrailing) {
		t.Errorf("parsing a trailing separator let it through")
	}
}

File: fixtures/adversarial/T6/ts/silent/after/src/format.test.ts
import { parse, SeparatorError } from "./format";

describe("parse", () => {
  it("refuses an empty input", () => {
    expect(() => parse("")).toThrow("input is empty");
  });

  it("refuses a stray separator", () => {
    expect(() => parse(";;")).toThrow(SeparatorError);
  });
});
